woocommerce/google-listings-and-ads
# Cleanup unused Google Ads services

### Changes proposed in this Pull Request:

The main purpose of this PR is to reduce the size of the Google Ads library. The large size of the library, and the many generated files cause problems on some hosts that timeout while installing/updating the plugin.
Since we only use a small part of the library this PR creates a cleanup script which removes services and enums from the library which we don't use.

Running the script reduces the size in the following way:
- Final zip file: 10.7MB > 7MB = 3.7MB smaller
- Files in zip file: 33.6MB > 22.2MB = 11.4MB smaller
- Googleads directory: 18.3MB > 8.2MB = 10.1MB smaller

![image](https://user-images.githubusercontent.com/11388669/167866430-b3b6e5c8-d434-4d6f-9d0e-642f748d6e11.png)

Resolves part of #314 

### Screenshots:
![image](https://user-images.githubusercontent.com/11388669/167582528-829ac443-ca96-4bc8-8351-b13ec491b224.png)

### Detailed test instructions:
1. Run composer install (which will run the cleanup script) 
2. Run the unit tests to confirm that all the needed files from the Ads library are still available
3. Run `npm run build`
4. Check the final zip file and confirm it's reduced in size

### Additional details:
The cleanup script is directly tied to the library version. If we update the library version the cleanup script will need to be modified and any newly added services will need to be added to the list. The list of services was taken from vendor/googleads/google-ads-php/src/Google/Ads/GoogleAds/Lib/V9/ServiceClientFactoryTrait.php

### Changelog entry
* Tweak - Cleanup unused Google Ads services.
